src/util/sample.h

```diff
@@ -35,17 +35,14 @@ class SampleUtil {
     // Sets every sample in pBuffer to zero
     inline
     static void clear(CSAMPLE* pBuffer, SINT numSamples) {
-        DEBUG_ASSERT(numSamples >= 0);
-        // Special case: This works, because the binary representation
-        // of 0.0f is 0!
-        memset(pBuffer, 0, sizeof(*pBuffer) * numSamples);
-        //fill(pBuffer, CSAMPLE_ZERO, iNumSamples);
+        fill(pBuffer, CSAMPLE_ZERO, numSamples);
     }
```

Comment:
fyi, this is [only equally performant on `-O2` and not on `-O1` on gcc](https://quick-bench.com/q/tYpHlnq8rlRf3aNihqCQC3oetwU). I guess thats fine? idk about msvc but afaikt their optimizer is terrible...

The original bug was saying that the `memset` is operating on wayy too much memory right? Maybe thats a hint that there is actually a bug here somewhere (maybe passing some negative `numSamples` and then underflowing)?

Reply:
I was also wondering if these was a reason for using memset. The original code already had the `fill` call commented out. Not sure if `fill` was the original implementation and turned out to be too slow, so that it was replaced by `memset`? Unfortunately there is no comment to explain it.

Reply:
Fun fact on clang 17 both Implementations are equivalent even with `-O1`.

Reply:
https://travisdowns.github.io/blog/2020/01/20/zero.html

Reply:
I can remember that fill was slow with MSVC. With GCC, memset even used AVX registers if it finds some on the target hardware. 

I have just checked with GCC and it turn out that fill also calls memset. 
In addition the fill has two extra checks that skips memset call if numSamples is zero or numSamples * sizeof(*pBuffer) is zero. They are redundant in our case. 

What is the warning you are trying to fix?  

I think we should test the current situation with MSVC before doing this change.

Reply:
I have noticed that 
```
void EngineDeck::processStem(CSAMPLE* pOut, const int iBufferSize) {
``` 
uses int and not SINT. Does that fix the warning?

Reply:
The diff value the compilers is complaining about is 0x1FFFFFFFC. Strange.

Reply:
The whole clear call is redundant at that place. 
Can we just replace it and keep the clear() implementation unchanged?

Reply:
Yes, let's the clear implementation as it is and silence this false positive. This is proven code and fast under all conditions, while fill is only fast under some conditions, when it internally calls memset.

Reply:
> https://travisdowns.github.io/blog/2020/01/20/zero.html

lol, it's actually faster with GCC and `-O1`: https://quick-bench.com/q/H1hlRRCNe_t-6-iRgTplPpVMKQA